### PR TITLE
Added Krush65 Solder ver

### PR DESCRIPTION
**Pull request by MaiTheSan**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sawnsprojects/krush65/solder/info.json [hunk 1]
```diff
@@ -0,0 +1,301 @@
+{
+    "keyboard_name": "Krush65 Solder",
+    "url": "",
+    "maintainer": "MaiTheSan",
+    "width": 16,
+    "height": 5,
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/info.json:6]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker/keymap.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker/keymap.c:83]: Please remove these, as these would be best in the keyboard's c file. 
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker_split_all/keymap.c [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker_split_all/keymap.c:83]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker_split_bs/keymap.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker_split_bs/keymap.c:84]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/default/keymap.c:83]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
+                if (clockwise) {
+                    tap_code(KC_EQL);
+                } else {
+                    tap_code(KC_MINS);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/via/keymap.c:96]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/rules.mk [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/sawnsprojects/krush65/solder/solder.c [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+void led_set_kb(uint8_t usb_led) {
+	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+  if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+    writePinLow(F0);
+  } else {
+    writePinHigh(F0);
+  }
+
+	led_set_user(usb_led);
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/solder.c:34]:
```suggestion
    /*  Custom encoder control - handles CW/CCW turning of encoder
     *  Default behavior:
     *    main layer:
     *       CW: move mouse right
     *      CCW: move mouse left
     *    other layers:
     *       CW: = (equals/plus - increase slider in Adobe products)
     *      CCW: - (minus/underscore - decrease slider in adobe products)
     */
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        switch (get_highest_layer(layer_state)) {

#ifdef MOUSEKEY_ENABLE
            case 0:
                // main layer - move mouse right (CW) and left (CCW)
                if (clockwise) {
                    tap_code(KC_MS_R);
                } else {
                    tap_code(KC_MS_L);
                }
                break;
#endif
            default:
                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
                if (clockwise) {
                    tap_code(KC_EQL);
                } else {
                    tap_code(KC_MINS);
                }
                break;
        }
    }
    return true;
}
```

File: keyboards/sawnsprojects/krush65/solder/config.h [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+#define MATRIX_COL_PINS { C7, C6, B6, B5, B4, D7, D5, D3 }
+#define UNUSED_PINS
+
+/* indicator */
+//  #define LED_CAPS_LOCK_PIN F0
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* rotary */
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/config.h:42]:
```suggestion
#define LED_CAPS_LOCK_PIN F0

/* rotary */
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker/keymap.c [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+
+
+void keyboard_post_init_user(void) {
+    // Customise these values to desired behaviour
+    //debug_enable = true;
+    //debug_matrix = true;
+    //debug_keyboard = true;
+    //debug_mouse = true;
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/ansi_blocker/keymap.c:50]:
```suggestion
```

File: keyboards/sawnsprojects/krush65/solder/keymaps/sawns/keymap.c [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+                    unregister_code(KC_LCTL);
+                } else {
+                    register_code(KC_LCTL);
+                    tap_code(KC_MINS);
+                    unregister_code(KC_LCTL);
+                }
+                break;
+        }
+    }
+}
```
Comment (drashna) [inline keyboards/sawnsprojects/krush65/solder/keymaps/sawns/keymap.c:109]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
    if (index == 0) {
        switch (get_highest_layer(layer_state)) {
            case 0:
                // main layer - move mouse right (CW) and left (CCW)
                if (clockwise) {
                    tap_code_delay(KC_VOLU, 10);
                } else {
                    tap_code_delay(KC_VOLD, 10);
                }
                break;

            default:
                // other layers - =/+ (quals/plus) (CW) and -/_ (minus/underscore) (CCW)
                if (clockwise) {
                    tap_code16(C(KC_EQL));
                } else {
                    tap_code(C(KC_MINS));
                }
                break;
        }
    }
    return false;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by MaiTheSan** [comment]
repull in https://github.com/qmk/qmk_firmware/pull/15063

**Pull request closed without merging**